app/grandchallenge/verifications/models.py

```diff
@@ -80,6 +83,29 @@ def save(self, *args, **kwargs):
 
         super().save(*args, **kwargs)
 
+        if self.has_changed("is_verified") and self.is_verified:
+            permission_request_classes = [
+                apps.get_model(
+                    app_label="algorithms",
+                    model_name="AlgorithmPermissionRequest",
+                ),
+                apps.get_model(
+                    app_label="archives", model_name="ArchivePermissionRequest"
+                ),
+                apps.get_model(
+                    app_label="reader_studies",
+                    model_name="ReaderStudyPermissionRequest",
+                ),
+            ]
+
+            for request_class in permission_request_classes:
+                pending_requests = request_class.objects.filter(
+                    user=self.user, status=request_class.PENDING
+                )
+
+                for request in pending_requests:
+                    process_access_request(request_object=request)
```

Comment:
Why do you need to process each request? Why not batch update them all to accepted?

Reply:
Do you mean update the model `status` value to True through ORM?
The `process_access_request` function does two other things besides status update, a call to `follow()`and a notification `send()`
Are those not needed in this case?

Reply:
You should be able to answer that yourself by reading `process_access_request`.

Reply:
I did read the function. The follow function allows the object activities to appear in the user's stream and this is applied to every request in general. If we opt for changing only the status value in this case, that means some requests are not being handled in the same way across the app.
My question still, do you want to keep those function calls for this case or not?

Reply:
Please read the function again because you clearly did not read it closely enough.

No notifications are generated if `request_object.base_object.access_request_handling == AccessRequestHandlingOptions.ACCEPT_VERIFIED_USERS` and `request_object.user.verification.is_verified`.

Notifications are generated for other things, but you should not be touching those requests in the first place, change it!

Reply:
Also - WHO received those requests? It's the admins of the challenge/reader study/archive. Why do they need to get notified AGAIN that someone request access when their verification status changed? It has nothing to do with them, they've already been notified and their verification status is irrelevant.

Reply:
Now I see the bigger picture. Thank you for the clear reasoning and explanation 👍